### ISSUE TITLE
Use options as optional not required arguments

### ISSUE THREAD

**Issue by pbabinca** [release-engineering/releng-sop#30]
See help of `koji-create-package-in-release`:

```
koji-create-package-in-release --help
usage: koji-create-package-in-release [-h] [--commit] --owner OWNER
                                      [--env ENV] [-d]
                                      RELEASE_ID PACKAGE [PACKAGE ...]

Create packages in a koji tag that maps to given release.

positional arguments:
  RELEASE_ID     PDC release ID, for example 'fedora-24', 'fedora-24-updates'.
  PACKAGE        Koji package, for example 'bash', 'kernel'.

optional arguments:
  -h, --help     show this help message and exit
  --commit       Program performs a dry-run by default. Enable this option to
                 apply the changes. (default: False)
  --owner OWNER  Package owner
  --env ENV      Select environment in which the program will make changes.
                 (default: default)
  -d, --debug    Print traceback for exceptions. By default only exception
                 messages are displayed. (default: False)

```

`--owner` is listed under _optional arguments_ even though in the usage it seems (and it indeed is) to be required. Please don't use this approach as _options_ are by its definition _optional_.

I believe _owner_ should be here as on of the positional arguments.

Related guidelines:
- [POSIX guidelines](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html)
- [GNU Coding Standards: Command-Line Interfaces](https://www.gnu.org/prep/standards/html_node/Command_002dLine-Interfaces.html#Command_002dLine-Interfaces)
- [The Art of Unix Programming: Command-Line Options](http://www.catb.org/esr/writings/taoup/html/ch10s05.html)

Answers to common arguments why there are _required options_:

"I'd like to make it optional later." - You'll break the API anyway so you should do that with major change of the version and indicate that as a such in release notes.

"I'd like to indicate type of the argument." - These are _options_ not for example keywords in python.

**Comment by csomh** [comment]
Good points!

Would it makes sense to have a(n optional) `koji_owner` setting in the environments file, which would be used if `--owner` is omitted?

I am thinking if it would be possible to change this to work similar to git's `clean.requireForce` setting, which if `true`, one of three options is required to be specified for the command -- but I am not sure, how much sense this would make from an end-user POV (i.e if there is a one-to-one mapping between and environment and owner).

**Comment by pbabinca** [comment]
As I understand how the option is used it isn't dependent on the environment. *owner* is package and product-specific.

Unless I'm missing something this doesn't solve the issue I'm having with the interface.

- You would still have this optional *but* on the top of that CLI would depend on some (via help) hidden configuration option which changes behaviour of the interface.
- Or you can read *some* config file even before displaying help but that *may* cause other issues (like where to look for config, what environment to actually show) and I don't recommend that.

**Comment by csomh** [comment]
@dmach, what do you think?
If we make `OWNER` an operand, which would be the preferred order?
`RELEASE_ID OWNER PACKAGE [PACKAGE ...]` or
`OWNER RELEASE_ID PACKAGE [PACKAGE ...]`

**Comment by dmach** [comment]
Fixed in 693e36799c49919911dcd12d5e6fa72b1782686f
